### PR TITLE
Tooltip: Improved Heatmap tooltip

### LINKED ISSUE #75843: Tooltip: Improved Heatmap tooltip
[Figma link](https://www.figma.com/file/tW8PMdzLb7hYYyBpb2xaYR/Universal-Tooltip?type=design&node-id=238-2638&mode=design&t=sDsktJ1imHOLLtdO-0)

Screenshot 2023-10-02 at 1 38 39 PM

Screenshot 2023-10-02 at 1 38 54 PM

### PR DESCRIPTION
**Pull request by adela-almasan**
Heatmap tooltip using `TooltipPlugin2`

Blocked by
[Dashboard: Overflow and z-index adjustments](https://github.com/grafana/grafana/pull/75597) ✅ 
[Tooltip: VizTooltip components #75794](https://github.com/grafana/grafana/pull/75794) ✅ 

https://github.com/grafana/grafana/assets/88068998/fa68285e-22b0-4e61-8d68-5ca8b0e161ca

Fixes #75843 
Fixes #76131

Test file
[native_histogram_heatmap.json.txt](https://github.com/grafana/grafana/files/12874312/debug-Native-2023-10-06.21.52.20.json.txt)

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/heatmap/tooltip/HeaderLabel.tsx
Comment (adela-almasan) [inline public/app/plugins/panel/heatmap/tooltip/HeaderLabel.tsx:1]: most of what's under heatmap/tooltip are part of a https://github.com/grafana/grafana/pull/75794. Will rebase after it gets merged.

File: public/app/plugins/panel/heatmap/HeatmapPanel.tsx
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/HeatmapPanel.tsx:136]: do we have a follow-up GH issue to track implementing hover value functionality?

File: public/app/plugins/panel/heatmap/module.tsx
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/module.tsx:426]: Is this still necessary?

```suggestion
```

File: public/app/plugins/panel/heatmap/tooltip/HeatmapTooltip.tsx
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/tooltip/HeatmapTooltip.tsx:349]: will initial version of heatmap tooltip support annotations?

File: public/app/plugins/panel/heatmap/tooltip/tooltipUtils.ts
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/tooltip/tooltipUtils.ts:1]: nit: but we probably could rename this file to just `utils` given it is in the `tooltip/` directory

**Review by nmarrs — COMMENTED** [review]
Will take another look when https://github.com/grafana/grafana/pull/75794 is merged - overall this is looking great!

File: packages/grafana-ui/src/components/uPlot/plugins/index.ts
Comment (adela-almasan) [reply packages/grafana-ui/src/components/uPlot/plugins/index.ts:4]: yes.. we'll do a cleanup at the very end..

File: public/app/plugins/panel/heatmap/HeatmapPanel.tsx
Comment (adela-almasan) [reply public/app/plugins/panel/heatmap/HeatmapPanel.tsx:136]: Here: https://github.com/grafana/grafana/issues/75879

File: public/app/plugins/panel/heatmap/tooltip/HeatmapTooltip.tsx
Comment (nmarrs) [reply public/app/plugins/panel/heatmap/tooltip/HeatmapTooltip.tsx:350]: jk - heatmaps dont currently support annotation if I remember correctly so disregard this question 😬

**Review by imatwawana — APPROVED** [review]
LGTM!

File: .betterer.results
Comment (adela-almasan) [inline .betterer.results:7421]: I will try to clean up the anys..

File: public/app/plugins/panel/heatmap/tooltip/utils.ts
Comment (adela-almasan) [inline public/app/plugins/panel/heatmap/tooltip/utils.ts:178]: I'm not completely sure these are correct. cc @leeoniya

**Comment by nmarrs** [comment]
We are considering adding a FF for tooltip stuff to go behind- with that in place and this behind that new FF this should be good for final review / merge :)

File: public/app/plugins/panel/heatmap/module.tsx
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/module.tsx:409]: We should hide this behind FF :)

File: public/app/plugins/panel/heatmap/HeatmapPanel.tsx
Comment (nmarrs) [reply public/app/plugins/panel/heatmap/HeatmapPanel.tsx:180]: Probably should remove legend changes so that we don't regress current legend behavior

**Review by nmarrs — COMMENTED** [review]
mob code review 😬

File: public/app/plugins/panel/heatmap/HeatmapPanel.tsx
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/HeatmapPanel.tsx:229]: are we confident releasing updated ZoomPlugin as part of this (i.e. not behind the FF?)

File: docs/sources/panels-visualizations/visualizations/heatmap/index.md
Comment (nmarrs) [inline docs/sources/panels-visualizations/visualizations/heatmap/index.md:117]: This UI option will only show up if FF is enabled - maybe we should make note of that here (I think there are examples of this in the docs somewhere 🤔)

File: .betterer.results
Comment (nmarrs) [reply .betterer.results:7041]: taking a 👀 at this :)

**Review by nmarrs — APPROVED** [review]
a few minor questions :) this is good overall though so I will approve and unblock you to merge when they are addressed 👍

File: public/app/features/visualization/data-hover/DataHoverView.tsx
Comment (leeoniya) [inline public/app/features/visualization/data-hover/DataHoverView.tsx:35]: this just pulls out the logic from the the component body so we can reuse it.

File: public/app/features/visualization/data-hover/DataHoverView.tsx
Comment (leeoniya) [inline public/app/features/visualization/data-hover/DataHoverView.tsx:99]: only change here is optional `padding` param at end, rest of diff is just logic extraction to exportable fn.

File: public/app/plugins/panel/heatmap/HeatmapHoverViewOld.tsx
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/HeatmapHoverViewOld.tsx:1]: this whole file is just a renamed of the unmodified `HeatmapHoverView.tsx`

**Pull request merged**
